### DISCUSSION TITLE
This is a start

### DISCUSSION THREAD

**Discussion by wu-clan** [fastapi-practices/fastapi_best_architecture_ui_arco#1] [Ideas]
- When to start? 
- Which techniques should be used?

**Comment by wu-clan** [comment]
Hi, @downdawn 

Do you have any ideas?

**Reply by downdawn** [reply]
[https://pro.antdv.com/](https://pro.antdv.com/)

I have compared several open-source front-end frameworks from major companies, and only this one incorporates the latest technology stack and is free. 

In fact, I have also made a small demo myself and I feel it is quite good.

**Reply by wu-clan** [reply]
Looks good, maybe as long as open source, recent updates, github stars a lot, it be OK.

In fact, my real idea is that it must look good.

**Reply by downdawn** [reply]
Yes, Arco supports theme customization.

[https://arco.design/vue/docs/theme](https://arco.design/vue/docs/theme)

**Reply by wu-clan** [reply]
Hi, @downdawn 

I have little experience in frontend, so I am not very sure about this direction

**Reply by jundaychan** [reply]
好看就选arco吧，用了arco 回不去了，哈哈

**Reply by jundaychan** [reply]
我用django（dva）这个项目和arco-design-vue3开发了前端，如果需要我可以提供出来

**Reply by wu-clan** [reply]
Of course, you are very welcome to join

**Reply by jundaychan** [reply]
咋加你，微信？

**Reply by wu-clan** [reply]
如果是开源项目可以直接在这里粘贴地址，我们一般在 Github 上面进行互动

**Comment by downdawn** [comment]
Hi, @wu-clan 

I use arco to implement simple login, logout, and query login log functions.

frontend: [https://github.com/downdawn/fastapi_best_architecture_ui](https://github.com/downdawn/fastapi_best_architecture_ui)
backend: [https://github.com/downdawn/fastapi_best_architecture/tree/adapt_to_frontend](https://github.com/downdawn/fastapi_best_architecture/tree/adapt_to_frontend)

![image](https://github.com/fastapi-practices/fastapi_best_architecture_ui/assets/41266749/5d2b3a92-e896-46a1-84a9-1854626b237a)

sys_menu.sql
```
INSERT INTO `sys_menu` (`id`, `name`, `level`, `sort`, `icon`, `path`, `menu_type`, `component`, `perms`, `status`, `remark`, `parent_id`, `created_time`, `updated_time`, `meta`) VALUES (2, 'dashboard', 0, 0, '', '/dashboard', 0, NULL, NULL, 1, NULL, NULL, '2023-06-27 16:54:56', NULL, '{\"icon\": \"icon-dashboard\", \"order\": 0, \"locale\": \"menu.server.dashboard\", \"requiresAuth\": true}');
INSERT INTO `sys_menu` (`id`, `name`, `level`, `sort`, `icon`, `path`, `menu_type`, `component`, `perms`, `status`, `remark`, `parent_id`, `created_time`, `updated_time`, `meta`) VALUES (3, 'Workplace', 0, 0, NULL, 'workplace', 0, NULL, NULL, 1, NULL, 2, '2023-06-27 16:58:02', NULL, '{\"locale\": \"menu.server.workplace\", \"requiresAuth\": true}');
INSERT INTO `sys_menu` (`id`, `name`, `level`, `sort`, `icon`, `path`, `menu_type`, `component`, `perms`, `status`, `remark`, `parent_id`, `created_time`, `updated_time`, `meta`) VALUES (4, 'arcoWebsite', 0, 0, NULL, 'https://arco.design', 0, NULL, NULL, 1, NULL, NULL, '2023-06-27 16:58:55', NULL, '{\"icon\": \"icon-link\", \"order\": 8, \"locale\": \"menu.arcoWebsite\", \"requiresAuth\": true}');
INSERT INTO `sys_menu` (`id`, `name`, `level`, `sort`, `icon`, `path`, `menu_type`, `component`, `perms`, `status`, `remark`, `parent_id`, `created_time`, `updated_time`, `meta`) VALUES (5, 'log', 0, 0, NULL, '/log', 0, NULL, NULL, 1, NULL, NULL, '2023-06-28 10:04:04', NULL, '{\"icon\": \"icon-bug\", \"order\": 2, \"locale\": \"menu.log\", \"requiresAuth\": \"true\"}');
INSERT INTO `sys_menu` (`id`, `name`, `level`, `sort`, `icon`, `path`, `menu_type`, `component`, `perms`, `status`, `remark`, `parent_id`, `created_time`, `updated_time`, `meta`) VALUES (6, 'Login', 0, 0, NULL, 'login', 0, NULL, NULL, 1, NULL, 5, '2023-06-28 10:05:53', NULL, '{\"roles\": [\"*\"], \"locale\": \"menu.log.login\", \"requiresAuth\": true}');
INSERT INTO `sys_menu` (`id`, `name`, `level`, `sort`, `icon`, `path`, `menu_type`, `component`, `perms`, `status`, `remark`, `parent_id`, `created_time`, `updated_time`, `meta`) VALUES (7, 'faq', 0, 0, NULL, 'https://arco.design/vue/docs/pro/faq', 0, NULL, NULL, 1, NULL, NULL, '2023-06-28 14:02:59', NULL, '{\"icon\": \"icon-question-circle\", \"order\": 9, \"locale\": \"menu.faq\", \"requiresAuth\": true}');

```

**Reply by downdawn** [reply]
This is just a simple start. If we want to continue developing with Arco, there are still many things that need to be adapted, with the most important ones being the menu and permissions.

[https://arco.design/vue/docs/pro/permission](https://arco.design/vue/docs/pro/permission)

**Reply by wu-clan** [reply]
Looks good, can you PR?

**Reply by downdawn** [reply]
#2 
https://github.com/fastapi-practices/fastapi_best_architecture/pull/162

**Reply by wu-clan** [reply]
Ok, already started.
